### PR TITLE
Output structure

### PR DESCRIPTION
**Pull request by jbathmann**
Adresses #166

### CODE REVIEW

File: TreeOutputLib/TreeOutput.py [hunk 1]
```diff
@@ -53,14 +99,83 @@ def iniOneFile(self, args):
...
+        elif tmp.text.strip() == "":
+            raise KeyError("Key '" + key + "' in project file at position " +
+                           "MangaProject__tree_output needs to be specified.")
+        return tmp.text
+
+    ## Writes output to predefined folder
+    #  For each timestep a file is created throughout the simulation.
+    #  This function is only able to work, if the output directory exists and
+    #  is empty at the begin of the model run
+    def writeOutput(self, tree_groups, time):
```
[the remark this answers is not in the document]
Comment (jbathmann) [reply TreeOutputLib/TreeOutput.py:165]: Thanks for the suggestion. I implemented the idea.

File: TreeOutputLib/TreeOutput.py [hunk 2]
```diff
@@ -53,14 +99,83 @@ def iniOneFile(self, args):
         from .OneFile import OneFile
```
[the remark this answers is not in the document]
Comment (jbathmann) [reply TreeOutputLib/TreeOutput.py]: Removed it

File: TreeOutputLib/TreeOutput.py [hunk 3]
```diff
@@ -4,63 +4,151 @@
...
-            self.iniNONE(args)
-        elif self.case == "OneTreeOneFile":
-            self.iniOneTreeOneFile(args)
-        elif self.case == "OneTimestepOneFile":
-            self.iniOneTimestepOneFile(args)
-        elif self.case == "OneTimestepOneFilePerGroup":
-            self.iniOneTimestepOneFilePerGroup(args)
-        elif self.case == "OneFile":
-            self.iniOneFile(args)
+        ## Directory, where output is saved. Please make sure it exists and is
```
Comment (mcwimm) [inline TreeOutputLib/TreeOutput.py:15]: This is not neccessary, right? The directory is created if it doesn't exist and with the allow_previous_... tag it does not need to be empty.

File: test/smallTests/testSetupsWithoutOGS/smallSetup_5.xml [hunk 4]
```diff
@@ -6,7 +6,7 @@
             <type> SimpleTest </type>
         </aboveground_competition>
         <belowground_competition>
-            <type> SimpleNetwork </type>
+            <type> SimpleTest </type>
```
Comment (mcwimm) [inline test/smallTests/testSetupsWithoutOGS/smallSetup_5.xml:9]: The file doesn't make sense but we will update all test files soon - so it doesn't matter, right?

**Comment by mcwimm** [comment]
Imo, ready to be merged!

File: TreeOutputLib/TreeOutput.py [hunk 5]
```diff
@@ -4,63 +4,151 @@
...
+        #  empty.
```
Comment (jbathmann) [inline TreeOutputLib/TreeOutput.py:16]:
```suggestion
        ## Directory, where output is saved. Please make sure it is empty
        #  or set allow_previous_output to true
```

**Comment by jbathmann** [comment]
Thanks for the review @mcwimm. I'd like for @jvollhueter's opinion before merging. :)

**Review by jvollhueter — COMMENTED** [review]
I think it's all very good. Especially the possibility to simply overwrite output now will make testing and developing new functionalities much easier. I have no suggestions for changes!

**Pull request merged**